dev/com.ibm.ws.security.openidconnect.client/src/com/ibm/ws/security/openidconnect/client/web/OidcRedirectServlet.java

```diff
@@ -167,8 +147,44 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
         }
     }
 
-    public boolean isRedirectionUrlValid(HttpServletRequest request, @Sensitive String requestUrl) {
-        return OidcClientUtil.isReferrerHostValid(request, requestUrl);
+    String getOriginalRequestUrl(HttpServletRequest request, HttpServletResponse response, String state) throws IOException {
+        //  when OIDCClientAuthenticatorUtil.doClientSideRedirect initially redirected the browser,
+        //  this cookie was set to hold the original URL.
+        //  Now it's time to get it back.
+        String cookieName = OidcStorageUtils.getOriginalReqUrlStorageKey(state);
+        String requestUrl = OidcClientUtil.getReferrerURLCookieHandler().getReferrerURLFromCookies(request, cookieName);
+        // 240540
+        //CookieHelper.clearCookie(request, response, cookieName, cookies); //clear the WAS_REQ_URL_OIDC cookie
+        OidcClientUtil.invalidateReferrerURLCookie(request, response, cookieName);
+        if (tc.isDebugEnabled() && requestUrl != null) {
+            Tr.debug(tc, "requestUrl: " + requestUrl);
+        }
+
+        if (requestUrl == null || requestUrl.isEmpty()) {
+            String errorMsg = Tr.formatMessage(tc, "OIDC_CLIENT_BAD_REQUEST_NO_COOKIE", request.getRequestURL()); // CWWKS1520E
+            Tr.error(tc, errorMsg);
+            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR); //will cause PublicFacingErrorServlet to run
+            return null;
+        }
+
+        if (!isRedirectionUrlValid(request, requestUrl, cookieName)) {
+            String errorMsg = Tr.formatMessage(tc, "OIDC_CLIENT_BAD_REQUEST_MALFORMED_URL_IN_COOKIE", request.getRequestURL(), (new URL(requestUrl)).getHost()); // CWWKS152XE
+            Tr.error(tc, errorMsg);
```

Comment:
Minor: Is `CWWKS152XE` meant the current `CWWKS1532E`?

Reply:
Correct, the OIDC_CLIENT_BAD_REQUEST_MALFORMED_URL_IN_COOKIE key maps to the CWWKS1532E message. I didn't notice the comment there when I refactored the code.